client/Contexts/Introspect/mock/expectations-basic.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import { Expectations } from '../ExpectationTypes';
+
+export default {
+  Other: {
+    type: 'Other',
+    fields: {
+      names: '[String]',
+      size: 'Int', // TODO how do we handle args?
+      roughSize: 'TShirtSize',
+      requiredRoughSize: 'TShirtSize!',
+      possibleRoughSizes: '[TShirtSize]',
+      requiredNames: '[String]!',
+      nonNullNames: '[String!]',
+      nonNullRequiredNames: '[String!]!',
+      topics: '[Topic]',
+    },
+  },
+  Topic: {
+    type: 'Topic', // the GraphQL type name
+    fields: {
+      name: 'String!', // the expected GraphQL type - could be string or object - which is checked if is available and thus query-able
+      partitions: 'Float',
+      replicas: 'Int',
+    },
+    operations: {
+      create: ({ mutations }) => {
+        return mutations['createTopic'] !== undefined;
+      }, // function that checks for a named mutation which provides the ability to create a topic for example
+      update: ({ mutations }) => {
+        return mutations['updateTopic'] !== undefined;
+      },
+      delete: ({ mutations }) => {
+        return mutations['deleteTopic'] !== undefined;
+      },
+      findByName: ({ queries }) => {
+        return (
+          queries['topicsByName'] !== undefined &&
+          queries['topicsByName'].args
+            .map((v) => (v.name === 'name' ? true : false))
+            .reduce((acc, curr) => {
+              return curr || acc ? true : false;
```

Comment:
This comment depends on whether I've interpreted the intention of the code.

If this interpretation is correct:

```
- Iterate over the list of args, mapping to `true|false` based off whether the arg is called `"name"`
- Reduce that mapped array into a single boolean, which will be `true` if any of the args were called `name`
```

If that is the case - I think the original code had a `return false` if the current value was false. That meant that if the first arg mapped to `true`, but the remaining `n` mapped to `false` - the reduce result would be `false` (I don't think you can short circuit a `reduce` so the first `true` would later be overridden by the `false`s).

In addition; if my description is correct, you can simplify it with the following:

```
queries['topicsByName'].args.findIndex(v => v.name === 'name') > -1
```

Reply:
I also note that this is now only in the mocks (but probably still valid in case this becomes a seed for the real expectations when later added back)